core/src/main/java/com/digitalpebble/stormcrawler/proxy/MultiProxyManager.java

```diff
@@ -0,0 +1,213 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * DigitalPebble licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.digitalpebble.stormcrawler.proxy;
+
+import com.digitalpebble.stormcrawler.protocol.httpclient.HttpProtocol;
+import com.digitalpebble.stormcrawler.util.ConfUtils;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.storm.Config;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Random;
+import java.util.Scanner;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * MultiProxyManager is a ProxyManager implementation for a multiple proxy endpoints
+ * */
+public class MultiProxyManager implements ProxyManager {
+    public enum ProxyRotation {
+        RANDOM,
+        LEAST_USED,
+        ROUND_ROBIN,
+    }
+
+    protected Random rng;
+    private SCProxy[] proxies;
+    private ProxyRotation rotation;
+    private final AtomicInteger lastAccessedIndex = new AtomicInteger(0);
+
+    private static final org.slf4j.Logger LOG = LoggerFactory
+            .getLogger(HttpProtocol.class);
+
+    /**
+     * Default constructor for setting up the proxy manager
+     * */
+    private void init(ProxyRotation rotation) {
+        // create rng with nano seed
+        this.rng = new Random(System.nanoTime());
+        // set rotation
+        this.rotation = rotation;
+    }
+
+    public MultiProxyManager() { }
+
+    @Override
+    public void configure(Config conf) throws FileNotFoundException, IllegalArgumentException {
```

Comment:
try catch and rethrow a RuntimeException instead. The calling class doesn;t need to know about the exact nature of the exception 
only this particular implementation expects a file, no need to have a specific exception about them